### PR TITLE
UI indicator performance

### PR DESCRIPTION
**Pull request by SAnatoliiS**

### CODE REVIEW

File: kit/controls/src/UIIndicator.tsx [hunk 1]
```diff
@@ -1,8 +1,17 @@
 import * as React from 'react';
 import type { StyleProp, ViewStyle } from 'react-native';
-import { MaterialIndicator } from 'react-native-indicators';
```
Comment (sertony) [inline kit/controls/src/UIIndicator.tsx:3]: Can we remove it from UIKit safely now? :)

Comment (sertony) [reply kit/controls/src/UIIndicator.tsx:3]: Oh, I guess not yet... But it's a good plan to do so in some near future ;)

File: kit/controls/src/UIIndicator.tsx [hunk 2]
```diff
@@ -11,20 +20,80 @@ export type UIIndicatorProps = {
...
+                },
+            ],
+        };
+    }, []);
+
+    const circumference = React.useMemo(() => size * Math.PI, [size]);
+
+    const animatedProps = useAnimatedProps(() => {
+        const strokeDashoffset =
+            circumference * (1 - progress.value * 2) * MAXIMUM_DASH_LENGTH_COEFFICIENT;
```
Comment (sertony) [inline kit/controls/src/UIIndicator.tsx:70]: Is it OK that it can become negative?

**Review by sertony — APPROVED** [review]

File: kit/controls/src/UIIndicator.tsx (hunk 2 above)
Comment (SAnatoliiS) [reply kit/controls/src/UIIndicator.tsx:70]: Yes, it should change from `circumference * MAXIMUM_DASH_LENGTH_COEFFICIENT` to `-circumference * MAXIMUM_DASH_LENGTH_COEFFICIENT`

**Review by sertony — APPROVED** [review]

**Review by savelichalex — APPROVED** [review]
Awesome!

**Pull request merged**